scripts/secure-boot/generate-images/generate-image.sh

```diff
@@ -0,0 +1,94 @@
+    openssl req -new -x509 -newkey rsa:2048 -subj "/CN=PK/" -keyout cert.key -out cert.crt -days 3650 -nodes -sha256
+
+    # generate the second key pair
+    openssl req -new -x509 -newkey rsa:2048 -subj "/CN=PK/" -keyout PK.key -out PK.crt -days 3650 -nodes -sha256
+
+    # make the CSR, its necessary to input something (country=PL, company=3mdeb)
+    openssl req -new -key cert.key -out cert.csr -subj "/C=PL/O=3mdeb"
+
+    # sign the CSR with the second key pair
+    # its necessary to `touch cert.ext`, otherwise it says its not there
+    touch cert.ext
+    openssl x509 -req -in cert.csr -CA PK.crt -CAkey PK.key -out cert.crt -CAcreateserial -extfile cert.ext
+    openssl x509 -in cert.crt -outform der -out cert.der
+
+elif [ "$TESTNAME" = "RSA" ]; then
+
+    openssl req -new -x509 -newkey rsa:$N -subj "/CN=PK/" -keyout cert.key -out cert.crt -days 3650 -nodes -sha256
+    openssl x509 -in cert.crt -outform der -out cert.der
+
+elif [ "$TESTNAME" = "ECDSA" ]; then
+
+    openssl req -new -x509 -newkey ec -pkeyopt ec_paramgen_curve:P-$N -subj "/CN=PK/" -keyout cert.key -out cert.crt -days 3650 -nodes -sha256
+    openssl base64 -d -in cert.crt -out cert.der
+
+elif [ "$TESTNAME" = "NOT_SIGNED" ]; then
+
+    # this image should contain just hello.efi, but since later we use sbsign
+    # with a cert, lets just create empty files so those later commands do not
+    # throw errors
+    touch cert.key cert.crt cert.der
+
+elif [ "$TESTNAME" = "GOOD_KEYS" ]; then
+
+    openssl req -new -x509 -newkey rsa:2048 -subj "/CN=PK/" -keyout cert.key -out cert.crt -days 3650 -nodes -sha256
+    openssl x509 -in cert.crt -outform der -out cert.der
+
+elif [ "$TESTNAME" = "BAD_KEYS" ]; then
+
+    # for this image, we just need to sign a hello.efi with a random cert, and
+    # later verify that its not allowed to be executed, while that random cert
+    # isnt enrolled
+    openssl req -new -x509 -newkey rsa:2048 -subj "/CN=PK/" -keyout cert.key -out cert.crt -days 3650 -nodes -sha256
+    openssl x509 -in cert.crt -outform der -out cert.der
+
+elif [ "$TESTNAME" = "BAD_FORMAT" ]; then
+
+    echo "this is not a certificate" > cert.crt
+    echo "this is not a certificate" > cert.der
+    echo "this is not a certificate" > cert.key
+
+elif [ "$TESTNAME" = "EXPIRED" ]; then
+
+    faketime '10 days ago' openssl req -new -x509 -newkey rsa:2048 -subj "/CN=ExpiredCert/" -keyout cert.key -out cert.crt -days 1 -nodes -sha256
+    openssl x509 -in cert.crt -outform der -out cert.der
+
+fi
+
+# sign hello.efi
+cp "$SCRIPTDIR"/hello.efi .
+sbsign --key cert.key --cert cert.crt --output signed-hello.efi hello.efi
+
+# copy all things to the image
+mkdir -p mnt/
+sudo mount -o loop image.img mnt/
+# simple check to see if it works
+echo "this image was created automatically" > check.txt
+# copy everything to the image (except for the image itself)
+sudo find ./ -maxdepth 1 -mindepth 1 ! -name 'image.img' ! -name 'mnt/' -exec cp {} mnt/ \;
+sudo umount image.img
```

Comment:
@WiktorG351 I'm not sure using sudo inside test is good idea, right now we would need to write password each time.
You could change it to:

```
udisksctl loop-setup -f image.img
```

drop sudo from `find` and add $LABEL variable with image label because udisksctl mounts image at `/run/media/<username>/<label>` or in script `/run/media/$(whoami)/$LABEL`

Reply:
If you remove those 3 sudo then you could most likely remove them also in `Make Certificate Images For Tests` keyword and `wrapper.sh` file.

Reply:
turns out [0.2 seconds](https://github.com/Dasharo/open-source-firmware-validation/pull/192/commits/bce10d72eb89079d32fb6db71846782fa2f3e593#diff-00f53487a7d91b72da783fcf7286011bd2c964cb60496c9c6e607a55e9ae748dR95) is enough :)